windows/bleachbit.nsi

```diff
@@ -1,493 +1,508 @@
-
-;--------------------------------
-;Uninstaller Section
-
-UninstallText "$(UNINSTALL_TEXT)"
-
-Section "$(SECTION_UNINSTALL)" SECTION_UNINSTALL
-    Delete $INSTDIR\bleachbit.exe.log
-
-    !ifndef NoTranslations
-      !include LocaleToUninstall.nsh
-    !endif
-
-    !include FilesToUninstall.nsh
-
-    DeleteRegKey HKCU "Software\${prodname}"
-    # delete normal shortcuts
-    RMDir /r "$SMPROGRAMS\${prodname}"
-    # delete any extra shortcuts
-    Delete "$DESKTOP\BleachBit.lnk"
-    Delete "$QUICKLAUNCH\BleachBit.lnk"
-    Delete "$SMSTARTUP\BleachBit.lnk"
-    # Remove Windows Explorer Shell Extension (Shredder)
-    DeleteRegKey HKCR "${SHRED_REGEX_KEY}"
-
-    # Remove the uninstaller as the very last step.
-    # If something goes wrong, let the user run it again.
-    Delete $INSTDIR\uninstall.exe
-    RMDir /REBOOTOK "$INSTDIR\."
-
-    # Remove the uninstaller from registry as the very last step.
-    # If something goes wrong, let the user run it again.
-    !insertmacro MULTIUSER_RegistryRemoveInstallInfo
-SectionEnd
-
-
-;--------------------------------
-;Uninstaller Functions
-
-Function un.onInit
-
-  !insertmacro MULTIUSER_UNINIT
-
-  !insertmacro MUI_UNGETLANGUAGE
-
-FunctionEnd
-
+;  vim: ts=4:sw=4:expandtab
+;
+;  BleachBit
+;  Copyright (C) 2008-2021 Andrew Ziem
+;  https://www.bleachbit.org
+;
+;  This program is free software: you can redistribute it and/or modify
+;  it under the terms of the GNU General Public License as published by
+;  the Free Software Foundation, either version 3 of the License, or
+;  (at your option) any later version.
+;
+;  This program is distributed in the hope that it will be useful,
+;  but WITHOUT ANY WARRANTY; without even the implied warranty of
+;  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;  GNU General Public License for more details.
+;
+;  You should have received a copy of the GNU General Public License
+;  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+;  @scriptversion v3.0.1.1429
+;  @scriptdate 2019-12-01
+
+
+;--------------------------------
+;Pack header:
+
+; Compress installer exehead with an executable compressor.
+
+; Paths should be absolute to allow building from any location.
+; Note that your executable compressor should not compress the first icon.
+
+!ifdef packhdr
+  ;Using UPX path info from setup_py2exe.py ->
+  !packhdr "$%TEMP%\exehead.tmp" '"\upx\upx.exe" -9 -q "$%TEMP%\exehead.tmp"'
+!endif
+
+
+;--------------------------------
+;Include Modern UI
+
+  !include "MUI2.nsh"
+
+
+;--------------------------------
+;General
+
+  ;Name and file
+  !define prodname "BleachBit"
+  !define COMPANY_NAME "BleachBit" ; # used by NsisMultiUser
+  !define PROG_AUTHOR "Andrew Ziem"
+  !define PROG_COPYRIGHT "Andrew Ziem"
+  BrandingText "${PROG_COPYRIGHT}"
+  Name "${prodname} ${VERSION}"
+
+  ; ----------------------------- disable warning overwrite default language strings (wrn 6030)
+  ;!pragma warning disable 6030

```

Comment:
It might be a good idea to push the warning and then after disabling the needed warning, pop back warning.

Reference: https://stackoverflow.com/questions/4193476/is-using-pragma-warning-push-pop-the-right-way-to-temporarily-alter-warning-lev

(Although the above is for C++, I guess the same would apply for NSIS, too)

Reply:
@sanjaykaria

The disabled warning  are only about missing strings in multiusernsh  file could be disabled AFTER will aligned all languages with all strings.
Otherwise you will have a lot of warnings that can create misunderstanding.

Reply:
@bovirus 
I understand the need of disabling the warning, but what am referring to is just saving the state of warning (in case it applies in NSIS too, as it does in C++) and then restoring the state back. 
I am not an expert in NSIS, but just a thought.

Reply:
@ sanjaykaria
For this moment to go ahead I decide to leave it as in original without warning but with comm,ent it hen if you need you can enable and remind the use.